src/openrct2-ui/windows/RideList.cpp

```diff
@@ -769,154 +775,141 @@ class RideListWindow final : public Window
-                        });
-                    break;
-                case INFORMATION_TYPE_QUEUE_TIME:
-                    currentListPosition = SortList(
-                        currentListPosition, rideRef, [](const Ride& thisRide, const Ride& otherRide) -> bool {
-                            return thisRide.GetMaxQueueTime() <= otherRide.GetMaxQueueTime();
-                        });
-                    break;
-                case INFORMATION_TYPE_RELIABILITY:
-                    currentListPosition = SortList(
-                        currentListPosition, rideRef, [](const Ride& thisRide, const Ride& otherRide) -> bool {
-                            return thisRide.reliability_percentage <= otherRide.reliability_percentage;
-                        });
-                    break;
-                case INFORMATION_TYPE_DOWN_TIME:
-                    currentListPosition = SortList(
-                        currentListPosition, rideRef, [](const Ride& thisRide, const Ride& otherRide) -> bool {
-                            return thisRide.downtime <= otherRide.downtime;
-                        });
-                    break;
-                case INFORMATION_TYPE_GUESTS_FAVOURITE:
-                    currentListPosition = SortList(
-                        currentListPosition, rideRef, [](const Ride& thisRide, const Ride& otherRide) -> bool {
-                            return thisRide.guests_favourite <= otherRide.guests_favourite;
-                        });
-                    break;
-                case INFORMATION_TYPE_EXCITEMENT:
-                    currentListPosition = SortList(
-                        currentListPosition, rideRef, [](const Ride& thisRide, const Ride& otherRide) -> bool {
-                            return thisRide.excitement <= otherRide.excitement;
-                        });
-                    break;
-                case INFORMATION_TYPE_INTENSITY:
-                    currentListPosition = SortList(
-                        currentListPosition, rideRef, [](const Ride& thisRide, const Ride& otherRide) -> bool {
-                            return thisRide.intensity <= otherRide.intensity;
-                        });
-                    break;
-                case INFORMATION_TYPE_NAUSEA:
-                    currentListPosition = SortList(
-                        currentListPosition, rideRef, [](const Ride& thisRide, const Ride& otherRide) -> bool {
-                            return thisRide.nausea <= otherRide.nausea;
-                        });
-                    break;
-            }
-            listIndex++;
+            RideListEntry entry{};
+            entry.Id = rideRef.id;
+            entry.Name = rideRef.GetName();
+
+            _rideList.push_back(entry);
+        }
+
+        switch (list_information_type)
+        {
+            case INFORMATION_TYPE_STATUS:
+                SortListByName();
```

Comment:
This seems to be the only function call, while the rest is inlined. Can't this also call the more generic `SortListByPredicate`?

Reply:
The name is cached in RideListEntry, the rest accesses the data over Ride so the predicate invocation is slightly different.